src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -455,43 +457,118 @@ def create_resource(properties,
+    return results[0] if len(results) == 1 else results
+
+
+def show_resource(resource_ids=None, resource_group_name=None,
                   resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
                   resource_name=None, api_version=None):
+    if resource_ids and isinstance(resource_ids, list):
+        return _call_on_ids(resource_ids, show_resource, api_version, "show")
+
     res = _ResourceUtils(resource_group_name, resource_provider_namespace,
                          parent_resource_path, resource_type, resource_name,
                          None, api_version)
     return res.get_resource()
 
 
-def delete_resource(resource_group_name=None,
+def delete_resource(resource_ids=None, resource_group_name=None,
                     resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
-                    resource_name=None, resource_id=None, api_version=None):
-    res = _ResourceUtils(resource_group_name, resource_provider_namespace,
-                         parent_resource_path, resource_type, resource_name,
-                         resource_id, api_version)
-    return res.delete()
-
-
-def update_resource(parameters,
+                    resource_name=None, api_version=None):
+    """
+    This function allows for the ability to delete multiple ids with dependencies on one another no matter the order.
+    This is done with multiple passes through the given ids.
+    A new pass is done as long as one resource was sucessfully deleted.
+    """
+    if not resource_ids or not isinstance(resource_ids, list):
+        res = _ResourceUtils(resource_group_name, resource_provider_namespace,
+                             parent_resource_path, resource_type, resource_name, None, api_version)
+        return res.delete()
+
+    from collections import deque
+    parsed_ids = deque()
+    for rid in resource_ids:
+        id_dict = parse_resource_id(rid)
+        if len(id_dict) == 1:
+            logger.error('az resource delete: error: argument --ids: invalid ResourceId value: \'%s\'' % rid)
+            return
+        parsed_ids.append(id_dict)
+
+    # queue stores operations so that they are run in parallel
+    operations = deque()
+    results = []
+    deleted = True
+    while parsed_ids and deleted:
+        deleted = False
+        for _ in range(len(parsed_ids)):
+            id_dict = parsed_ids.pop()
+            try:
+                operations.append(delete_resource(
+                    resource_group_name=id_dict['resource_group'],
+                    resource_provider_namespace=id_dict['resource_namespace'],
+                    parent_resource_path=id_dict['resource_parent'],
+                    resource_type=id_dict['resource_type'],
+                    resource_name=id_dict['resource_name'], api_version=api_version))
+                logger.debug("deleting", id_dict['resource_name'])
```

Comment:
Here you might want to reassemble the ID... `resource_id(id_dict)`.  A lot of times people will use the same name for resources.  You could have a vnet, public_ip, vm, etc all with the same name.